### DISCUSSION TITLE
Como identificar se eu/vc estamos acomodados com a tecnologia que já sabemos?

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#419] [Discussão]
Estou pensando nisso e gostaria de ver a opinião de vocês para ouvir pensamentos diferentes.

**Comment by joao-alberto** [comment]
Eu acho que o principal indicador é nem procurar outras maneiras de resolver determinado problema. Fazendo um paralelo, é semelhante a homeostase, ou a necessidade de um sistema se manter em ordem ou estável.

Como eu lutaria contra isso?

Tentando identificar pensamentos como esse:

> Eu já resolvi um problema semelhante a esse de tal maneira, logo não tenho porque mudar.

Se identificar que isso está ocorrendo, eu mudo, busco outra alternativa.

**Comment by brunoravanhani** [comment]
Na minha opinião, quando seu trabalho/projeto/job ficou fácil, tipo rotina de chão de fábrica.

**Comment by fabiorocha** [comment]
**TL;DR:** tente ser consciente sobre aquilo que você sabe e aquilo que você não sabe. Tente estar sempre evoluindo em relação àquela tecnologia, que é basicamente o trabalho de outra pessoa e que está em constante atualização. Tente contribuir com aquela tecnologia se você já a domina.

--

Puts velho. Tão boa a pergunta que resolvi compilar algumas das coisas que já li a respeito pra eu mesmo usar como lembrete no futuro.

Essa questão envolve tanta coisa que a gente tem que discutir alguns tópicos relacionados primeiro.

## Zona de Conforto

Acho que esse foi o primeiro conceito que me veio à mente quando você fez a pergunta, que pode ser traduzida para: "Como saber que estou na minha zona de conforto quanto à tecnologia X?".

A imagem abaixo [1] ilustra bem o que penso em relação a isso. Estamos na nossa zona de conforto em relação a alguma coisa quando já sabemos o suficiente sobre ela e aprendemos apenas o suficiente pra nos mantermos confortáveis (note que a zona de conforto está dentro da zona de aprendizado mas elas duas não são iguais).

Fora dessa zona existe a Zona de Aprendizado, que é quando vamos atrás daquilo que não sabemos.

<img src="http://fabiorocha.com/img/comfort.png" alt="Drawing" style="width:400px;"/>

O problema é que para irmos atrás do que não sabemos, nós precisamos admitir ou **saber que não sabemos sobre algo**. Aí entram os 4 estágios de qualquer aprendizado [2]:

1. Incompetência inconsciente: "Você não sabe que não sabe". Muita gente vive nessa etapa. Aprende um pouco de um assunto e acha que sabe o suficiente.

2. Incompetência consciente: "Você sabe que não sabe". Esse estágio é equivalente à zona de aprendizado aí de cima. Alguma coisa faz você descobrir que não sabe o suficiente. Nessa etapa você vai atrás do conhecimento que falta. O importante aqui é você ir atrás do conhecimento aos poucos e ir aumentando sua zona de conforto lentamente. Muita gente quer abraçar o mundo com as mãos nessa fase e vai direto pra "Zona de Perigo", que é onde você fica desmotivado com a quantidade de coisas que não sabe, ou acha a complexidade do que não sabe muito alta e... desiste.

3. Competência consciente: "Você sabe que sabe". Passando a fase acima, você já tem um auto-conhecimento e sabe exatamente o que sabe e o que não sabe. Aqui você já vive na zona de aprendizado e se atualiza constantemente.

4. Competência insconciente: "Você sabe tanto que acha que sabe menos do que sabe". Depois de um tempo na fase acima, você sabe tanto que às vezes fica díficil saber o que sabe. Às vezes acaba até achando que sabe menos do que realmente sabe (síndrome do impostor, assunto pra outra hora).

<img src="http://fabiorocha.com/img/competence.jpg" alt="Drawing" style="width:400px;"/>

## Tecnologia

O que é tecnologia? Quando você para pra pensar, tudo que nós usamos no dia a dia foi construído por pessoas como nós, e as tecnologias que usamos, **todas**, também são frutos do trabalho de alguém.

Steve Jobs uma vez falou: "A vida pode ser muito mais ampla, uma vez que você descobre um simples fato, e isso tudo que está ao seu redor que você chama de vida foi feita por pessoas que não eram mais inteligentes do que você. E você pode mudá-la, você pode influenciá-la, você pode construir as suas próprias coisas que outras pessoas podem usar. Uma vez que você aprender isso, você nunca será o mesmo novamente." [3] **Video:**


[![Steve Jobs' Vision of the World](https://i.ytimg.com/vi/UvEiSa6_EPA/hqdefault.jpg)](https://www.youtube.com/watch?v=UvEiSa6_EPA)

Quando você para pra pensar, as linguagens são feitas por pessoas (JavaScript, hoje, por exemplo é basicamente a implementação pelos browsers do padrão discutido pelo TC39), os browsers são feitos por pessoas, os sistemas operacionais, também feitos por pessoas, enfim: **pessoas, pessoas, pessoas**. Então tecnologia, como sabemos, é um alvo em constante movimento e se a gente não acompanhar a gente fica pra trás mesmo.


## Como identificar

Bom, depois de tantas voltas, vamos ao ponto X da questão. Como identificamos que estamos na nossa zona de conforto? Pra mim isso pode ser dividido em dois pontos:

### Consumo

Basicamente a gente consome tecnologia quando está aprendendo, através de artigos, tutoriais, livros, etc:

- Quando foi a última vez que me atualizei em relação a tecnologia X?
- Eu estou utilizando a última versão dessa tecnologia?
- Eu ainda estou aprendendo sobre X ou só utilizo aquilo que já sei?

### Criação

Como profissionais da área, também produzimos tecnologia, através de projetos. Nesse aspecto:

- Eu já implementei tal coisa N vezes. Posso fazer melhor dessa vez?
- Quem revisa o meu trabalho? Existe alguém com mais conhecimento que eu que pode dar feedback naquilo que eu estou produzindo?
- Eu preciso resolver um problema, vou utilizar a tecnologia X porque já sei ou porque é a melhor ferramenta pro trabalho?
- Eu utilizo essa tecnologia há bastante tempo, como eu faço pra mudar pro time que CRIA essa tecnologia? Como eu contribuo?


## Como se manter fora da zona de conforto

Assumindo que você quer viver fora da sua zona de conforto, que por definição **não é algo confortável**:

- Identifique aquilo que não sabe e aprenda, aos poucos.
- Identifique pessoas que sabem mais que você, tente aprender com elas.
- Conheça a comunidade ao redor daquela tecnologia que você gosta, tente participar.
- Tente contribuir com a criação da próxima versão da tecnologia.
- Revise seu modo de aprendizado: "O modo pelo qual aprendo é o mais apropriado?" Livros tendem a ter informações mais ultrapassadas, enquanto artigos ou listas de discussões como essas têm informações mais atualizadas. Pra tecnologias que não mudam tão rapidamente (ex: algoritmos), um bom livro ainda é um ótimo jeito de aprender.

Acho que é isso, a dica principal que eu daria é: esteja sempre revisando sua zona de conforto e tentando evoluir. É fácil se acomodar quando já sabemos o suficiente pra fazer o trabalho do dia a dia. Abs.


____
#### Referências

[1] [How Your Comfort Zone Is Sabotaging Your Success](http://www.huffingtonpost.com/kevin-kleitches/how-your-comfort-zone-is_b_5888764.html)

[2] [Four stages of competence](https://en.wikipedia.org/wiki/Four_stages_of_competence)

[3] [Steve Jobs' Vision of the World](https://www.youtube.com/watch?v=UvEiSa6_EPA)

**Comment by leandroruel** [comment]
wow... just... wow

**Comment by victormiguez** [comment]
@fabiorocha parabéns pela resposta, ficou sensacional!

**Comment by danilovaz** [comment]
@fabiorocha Já deixei como aba fixa aqui, sempre que surgir aquela bad de zona de conforto, vou vir aqui ler novamente :beers:

**Comment by ranosi** [comment]
Tópico bem legal, com certeza vale sempre reler. Na maioria do tempo costumo cair nesse problema aqui: "direto pra "Zona de Perigo", que é onde você fica desmotivado com a quantidade de coisas que não sabe, ou acha a complexidade do que não sabe muito alta e... desiste."

Isso acabou até me tirando da área por um tempo e hoje to correndo atrás do tempo perdido.
